client/client.go

```diff
@@ -371,6 +391,42 @@ func (w *WrapperV2) GetCluster(clusterID string, p *AuxiliaryParams) (*clusters.
 	return response, nil
 }
 
+// GetClusterStatus fetches status details from the
+// status endpoint (/v4/clusters/{clusterid}/status/).
+// Only a few details are returned.
+func (w *WrapperV2) GetClusterStatus(clusterID string) (*ClusterStatus, *http.Response, error) {
```

Comment:
I'd maybe prefer this rather than creating a separate client. Maybe add proper explanation for the Marshal-Unmarshal-dance in a comment so that someone who might wonder about this gets the story as well.

Reply:
Looks good to me. Yes, commenting for the future colleague trying to wrap her/his hand around this would be nice.